src/colvar_arithmeticpath.h

```diff
@@ -77,20 +103,15 @@ void ArithmeticPathBase<element_type, scalar_type, path_type>::computeValue() {
         for (size_t j_elem = 0; j_elem < num_elements; ++j_elem) {
             exponent_tmp += weights[j_elem] * frame_element_distances[i_frame][j_elem] * weights[j_elem] * frame_element_distances[i_frame][j_elem];
         }
-        exponent_tmp = exponent_tmp * -1.0 * lambda;
-        // prevent underflow if the argument of cvm::exp is less than -708.4
-        if (exponent_tmp > -708.4) {
-            exponent_tmp = cvm::exp(exponent_tmp);
-        } else {
-            exponent_tmp = 0;
-        }
-        numerator_s += static_cast<scalar_type>(i_frame) * exponent_tmp;
-        denominator_s += exponent_tmp;
-        s_numerator_frame[i_frame] = static_cast<scalar_type>(i_frame) * exponent_tmp;
-        s_denominator_frame[i_frame] = exponent_tmp;
+        exponents[i_frame] = exponent_tmp * -1.0 * lambda;
+        s_denominator_frame[i_frame] = cvm::exp(exponents[i_frame]);
+        s_numerator_frame[i_frame] = static_cast<scalar_type>(i_frame) * s_denominator_frame[i_frame];
+        numerator_s += s_numerator_frame[i_frame];
+        denominator_s += s_denominator_frame[i_frame];
     }
-    s = numerator_s / denominator_s * normalization_factor;
-    z = -1.0 / lambda * cvm::logn(denominator_s);
+    const auto log_s = cvm::logn(normalization_factor) + logsumexp(exponents, frame_indexes) - logsumexp(exponents);
+    s = cvm::exp(log_s);
+    z = -1.0 / lambda * logsumexp(exponents);
```

Comment:
Even if the overhead is small, I recommend caching `logsumexp(exponents)` instead of evaluating it twice in quick succession.